Cannot provide user-defined Debug implementation for enum

### Input C/C++ Header

```C++
typedef enum E {
  A = 0,
  B = 1
} ET;
```

### Bindgen Invocation

```Rust
fn main() {
    let bindings = bindgen::builder()
        .header("src/header.h")
        .default_enum_style(bindgen::EnumVariation::Rust { non_exhaustive: true })
        .derive_debug(false)  // Don't derive debug
        .parse_callbacks(Box::new(MyCallback))
        .generate()
        .expect("Unable to generate bindings");

    let out_path = std::path::Path::new("src/ffi.rs");
    bindings.write_to_file(out_path).expect("Couldn't write bindings");
}

#[derive(Debug)]
struct MyCallback;
impl core::panic::UnwindSafe for MyCallback {}

impl bindgen::callbacks::ParseCallbacks for MyCallback {
    fn add_derives(&self, name: &str) -> Vec<String> {
        let mut v: Vec<String> = Vec::new();
        if name == "E" {
            v = vec!["Debug".to_string()];  // Provide my own Debug impl somewhere
        }
        v
    }
}
```

### Actual Results

The above fails with:

```shell
error[E0119]: conflicting implementations of trait `std::fmt::Debug` for type `E`
 --> src/ffi.rs:5:51
  |
5 | #[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Debug)]
  |          ----- first implementation here          ^^^^^ conflicting implementation for `E`
```

The `.derive_debug(false)` is set, but a `Debug` impl is still being added, which prevents me from providing my own implementation somewhere (not necessarily in a callback).

### Expected Results

I'd like to be able to provide my own implementation of `Debug` for the type above, while still automatically adding the `Debug` trait to all other types.

--- a/src/codegen/mod.rs
+++ b/src/codegen/mod.rs
@@ -3005,14 +3005,9 @@ impl CodeGenerator for Enum {
 
         if !variation.is_const() {
             let mut derives = derives_of_item(item, ctx);
-            // For backwards compat, enums always derive Debug/Clone/Eq/PartialEq/Hash, even
-            // if we don't generate those by default.
-            if !item.annotations().disallow_debug() {
-                derives.insert(DerivableTraits::DEBUG);
-            }
-            if !item.annotations().disallow_copy() {
-                derives.insert(DerivableTraits::COPY);
-            }
+            // For backwards compat, enums always derive
+            // Clone/Eq/PartialEq/Hash, even if we don't generate those by
+            // default.
             derives.insert(
                 DerivableTraits::CLONE |
                     DerivableTraits::HASH |

--- a/tests/expectations/tests/enum-no-debug-rust.rs
+++ b/tests/expectations/tests/enum-no-debug-rust.rs
@@ -13,7 +13,7 @@ pub struct foo {
 pub const foo_FOO_A: foo__bindgen_ty_1 = foo__bindgen_ty_1::FOO_A;
 pub const foo_FOO_B: foo__bindgen_ty_1 = foo__bindgen_ty_1::FOO_B;
 #[repr(u32)]
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+#[derive(Copy, Clone, Hash, PartialEq, Eq)]
 pub enum foo__bindgen_ty_1 {
     FOO_A = 0,
     FOO_B = 1,
@@ -51,7 +51,7 @@ impl Default for foo {
     }
 }
 #[repr(u32)]
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+#[derive(Copy, Clone, Hash, PartialEq, Eq)]
 pub enum Foo {
     Bar = 0,
     Qux = 1,
@@ -70,7 +70,7 @@ pub enum NoDebug {
 }
 #[repr(u32)]
 /// <div rustbindgen derive="Debug"></div>
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq)]
+#[derive(Copy, Clone, Hash, PartialEq, Eq, Debug)]
 pub enum Debug {
     Debug1 = 0,
     Debug2 = 1,

